deblur/deblurring.py

```diff
@@ -0,0 +1,167 @@
+
+    if len(aligned_lengths) != 1 or len(unaligned_lengths) != 1:
+        raise ValueError(
+            "Not all sequence have the same length. Aligned lengths: %s, "
+            "sequence lengths: %s"
+            % (", ".join(map(str, aligned_lengths)),
+               ", ".join(map(str, unaligned_lengths))))
+
+    seqs = sorted(seqs, key=attrgetter('frequency'), reverse=True)
+    return seqs
+
+
+def deblur(input_seqs, read_error=0.05, mean_error=None, error_dist=None,
+           indel_prob=0.01, indel_max=3):
+    """Deblur the reads
+
+    Parameters
+    ----------
+    input_seqs : iterable of (str, str)
+        The list of input sequences in (label, sequence) format
+    read_error : float, optional
+        The read error rate. Default: 0.05
+    mean_error : float, optional
+        The mean error, used for original sequence estimate. Default: same
+        value as `read_error`
+    error_dist : list of float, optional
+        A list of error probabilities. The length of the list determines the
+        amount of hamming distances taken into account. Default: None, computed
+        from mean error and sequence length.
+    indel_prob : float, optional
+        Indel probability (same for N indels). Default: 0.01
+    indel_max : int, optional
+        The maximal number of indels expected by errors. Default: 3
+
+    Results
+    -------
+    list of Sequence
+        The deblurred sequences
+
+    Notes
+    -----
+    mean_error is used only for normalizing the peak height before deblurring,
+    whereas read_error is used for calculating the expected number of errors
+    for each position.
+    The array 'error_dist' represents the error distribution, where
+    Xi = max frequency of error hamming. The length of this array - 1 limits
+    the hamming distance taken into account, i.e. if the length if `error_dist`
+    is 10, sequences up to 10 - 1 = 9 hamming distance will be taken into
+    account
+    """
+    # Get the sequences
+    seqs = get_sequences(input_seqs)
+
+    # If mean error is not provided, use the same value as read_error
+    mean_error = mean_error if mean_error else read_error
+
+    # if error_list not supplied, use the default (22 mock mixture setup)
+    mod_factor = pow((1 - mean_error), seqs[0].unaligned_length)
+    if not error_dist:
+        error_dist = np.array(
+            [1.0 / mod_factor, pow(read_error, 1) / mod_factor, 0.01, 0.01,
+             0.01, 0.005, 0.005, 0.005, 0.005, 0.005, 0.005, 0.001, 0.001,
+             0.001, 0.001])
+    else:
+        error_dist = np.array(error_dist) / mod_factor
+
+    max_h_dist = len(error_dist)-1
+
+    for seq_i in seqs:
+        # no need to remove neighbors if freq. is <=0
+        if seq_i.frequency <= 0:
+            continue
+
+        # Correct for the fact that many reads are expected to be mutated
+        num_err = error_dist * seq_i.frequency
+        num_err = [val * seq_i.frequency for val in error_dist]
```

Comment:
Ups, yeah, I was making more numpythonic code and I forgot to remove the old one hahaha